src/widget/wlibrarytableview.cpp

```diff
@@ -100,33 +100,71 @@ void WLibraryTableView::moveSelection(int delta) {
             if (row + 1 < pModel->rowCount()) {
                 selectRow(row + 1);
             }
-
             delta--;
         } else {
             // i is negative, so we want to move the highlight up
             int row = current.row();
             if (row - 1 >= 0) {
                 selectRow(row - 1);
             }
-
             delta++;
         }
     }
 }
 
 void WLibraryTableView::saveVScrollBarPos(TrackModel* key){
+    qDebug()<<"";
+    qDebug()<<"    WLibraryTableView::saveVScrollBarPos";
+    qDebug()<<"       m_vScrollBarPosValues["<<key<<"] ="<<verticalScrollBar()->value();
     m_vScrollBarPosValues[key] = verticalScrollBar()->value();
+    if (currentIndex().isValid()) {
+        // Act only when there is a focused cell. For example, there is none
+        // when library features are added initially after start.
+        qDebug()<<"        QModelIndexList selection = selectionModel()->selectedRows();";
+        QModelIndexList selection = selectionModel()->selectedRows();
+        if (selection.size() > 0) {
+            m_selectedIndices[key] = selection;
+        }
+    }
 }
 
 void WLibraryTableView::restoreVScrollBarPos(TrackModel* key){
+    qDebug()<<"";
+    qDebug()<<"    WLibraryTableView::restoreVScrollBarPos";
     updateGeometries();
 
-    if (m_vScrollBarPosValues.contains(key)){
+    if (m_vScrollBarPosValues.contains(key)) {
+        qDebug()<<"   restore previous vertical scrollbar pos from m_vScrollBarPosValues["<<key<<"]";
         verticalScrollBar()->setValue(m_vScrollBarPosValues[key]);
-    }else{
+    } else {
+        qDebug()<<"   store & set default vertical scrollbar position";
         m_vScrollBarPosValues[key] = 0;
         verticalScrollBar()->setValue(0);
     }
+    if (m_selectedIndices.contains(key)) {
+        QModelIndexList selection = m_selectedIndices[key];
+        // See documentation of failed commands in https://github.com/mixxxdj/mixxx/pull/2378
```

Comment:
I think it is better to copy the whole explanation here.

Reply:
This spans around 20 lines, I thought it would bloat this function. ([description](https://github.com/mixxxdj/mixxx/pull/2378#issue-348474420), at the bottom) You're sure we need this here?